user-interface/src/main/java/life/qbic/datamanager/AppConfig.java

```diff
@@ -198,9 +216,14 @@ public ProjectRegisteredPolicy projectRegisteredPolicy(SampleCodeService sampleC
 
   @Bean
   public SampleRegisteredPolicy sampleRegisteredPolicy(
-      BatchRegistrationService batchRegistrationService, JobScheduler jobScheduler) {
+      BatchRegistrationService batchRegistrationService,
+      SampleInformationService sampleInformationService,
+      ExperimentInformationService experimentInformationService,
+      ProjectInformationService projectInformationService, JobScheduler jobScheduler) {
```

Comment:
why so many services?

Reply:
batch service to add the sample to the batch,
sample and experiment service to find the project id, which is not connected to the sample,
project service to update the project